src/pages/MenuPage.js

```diff
@@ -32,12 +76,90 @@ export class MenuPage extends React.Component {
     );
   }
 
+  componentWillReceiveProps(props) {
+    // eslint-disable-next-line react/prop-types
+    const { topRoute } = props;
+    if (topRoute) this.refreshData();
+  }
+
   componentWillUnmount() {
     const { database } = this.props;
 
     database.removeListener(this.databaseListenerId);
   }
 
+  refreshData = () => {
```

Comment:
if you go with the `Popover` wrapper, this logic isn't needed and you can just pass directly in props to each badge set `database.objects("CustomerInvoice").filtered("status != 'finalised'").length` - the component itself can take care of the `title` and obviously the `type`, then none of this state is needed in either `MenuPage` or `NavigationBar`.

Reply:
Refactored with 7b3ab4b